cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/WorkflowDataset.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.store;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.dataset.lib.AbstractDataset;
+import co.cask.cdap.api.dataset.table.Row;
+import co.cask.cdap.api.dataset.table.Scan;
+import co.cask.cdap.api.dataset.table.Scanner;
+import co.cask.cdap.api.dataset.table.Table;
+import co.cask.cdap.common.utils.ImmutablePair;
+import co.cask.cdap.data2.dataset2.lib.table.FuzzyRowFilter;
+import co.cask.cdap.proto.Id;
+import com.google.common.collect.ImmutableList;
+import com.google.common.primitives.Longs;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.google.inject.Inject;
+import org.apache.avro.generic.GenericData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ *
+ */
+public class WorkflowDataset extends AbstractDataset {
+
+  private static final Gson GSON = new Gson();
+  private static final String RUNID = "runId";
+  private static final String TIME_TAKEN = "timeTaken";
+  private static final String NODES = "nodes";
+  private static final String DELIMITER = ":";
+  private static final Logger LOG = LoggerFactory.getLogger(WorkflowDataset.class);
+
+  private final Table table;
+
+  @Inject
+  public WorkflowDataset(Table table) {
+    super("ignored", table);
+    this.table = table;
+  }
+
+  private byte[] createRowKey(String key, long time) {
+    byte[] result = Bytes.toBytes(key);
+    byte[] rowKey = new byte[result.length + 8];
```

Comment:
+1, also should comment about why you are adding size of long, saying we have startTimestamp at the end.